src/main/java/net/minecrell/serverlistplus/core/replacement/DefaultPatternPlaceholder.java

```diff
@@ -91,14 +91,14 @@ public Object next() {
                     group = matcher.group(2);
                     int max = group == null ? DEFAULT_LIMIT : Integer.parseInt(group);
                     group = matcher.group(3);
-                    String delimiter = group == null ? DEFAULT_DELIMITER : group;
+                    String[] delimiters = ((group == null) ? DEFAULT_DELIMITER : group).split("(?<!\\\\)\\|");
 
                     StringBuilder result = new StringBuilder();
                     int i = 1;
                     while (true) {
                         result.append(players.next());
                         if (i >= max || !players.hasNext()) break;
-                        result.append(delimiter); i++;
+                        result.append(delimiters[(i++ - 1) % delimiters.length]);
```

Comment:
This code looks rather weird, although this is mostly my fault. Can we try to refactor this to a `for` loop like:

```java
for (int i = 0; i < max && players.hasNext(); i++) {
    if (i != 0) result.append(delimiters[i % delimiters.length]);
    result.append(players.next());
}
```

Would be nice if you could test this, not sure if this is 100% correct.